### DISCUSSION TITLE
Welcome to SvelteKitAuth Discussions!

### DISCUSSION THREAD

**Discussion by Dan6erbond** [Dan6erbond/sk-auth#9] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by x-ror** [comment]
Do u have plans to provide a custom redirect_uri?

**Reply by Dan6erbond** [reply]
Are you referring to the `/api/auth` part of the scheme, or `/callback/<provider>`?

---

As for the base path of the auth class, it can be placed wherever you wish. The default is `/api/auth`, and if you place it, in, say `/auth` instead all you need to do is provide the `basePath` in `AuthConfig`:

```ts
interface AuthConfig {
  providers: Provider[];
  callbacks?: AuthCallbacks;
  jwtSecret?: string;
  jwtExpiresIn?: string | number;
  host?: string;
  basePath?: string;
}
```

## API Routes

The way providers work, though, with `signin()` and `callback()` is more or less given. There are considerations being made for allowing the `signin()` routes to set cookies using a `setCookie()` helper provided by the `Auth` class, as seen in my [comment](https://github.com/Dan6erbond/sk-auth/discussions/20#discussioncomment-778694) on #20 but the endpoint needs a way to know which part of the flow is being handled which means the `/signin/<provider>` and `/callback/<provider>` routes will only change should the need arise for a different interface.

Now that you bring that up, though, I have been thinking up a possible method to handle provider metadata with a `getMetadata()` function in the abstract class, as seen in #18, which would mainly be used for frontends to dynamically generate login pages, but it could open the possibility to define arbitrary routes, and mark them as `handler` and `redirect` where `handler` routes handle endpoints, and `redirect` routes are redirected to and return a user/redirect URL combo the way `callback()` currently is designed.

This might be something I'll add in a v2 or rewrite of the library, as it will require rethinking the way currently API routes are handled, since they're fairly restrictive at the moment.

**Reply by x-ror** [reply]
i mean about /callback/<provider>

**Reply by Dan6erbond** [reply]
In that case it's something I'll have to consider, if I can setup a mechanism for generating the provider schema, one could define arbitrary routes that aren't confined to the `/signin` and `/callback` schema anymore. 

I'll see about creating a dedicated issue for this tomorrow. Feel free to contribute so we can see about coming to a consensus on giving developers as much freedom as possible with SvelteKitAuth!

**Comment by x-ror** [comment]
@Dan6erbond which bundle of the icons is u using for google, Facebook, Reddit icon...
i want to add twitch

**Reply by Dan6erbond** [reply]
The icons in the demo app are taken from [Devicon](https://devicon.dev/).
